src/devices/bus/nes/mmc3.cpp

```diff
@@ -119,7 +119,14 @@ void nes_txrom_device::mmc3_common_initialize( int prg_mask, int chr_mask, int i
 {
 	m_mmc_prg_bank[0] = m_mmc_prg_bank[2] = 0xffe; // m_mmc_prg_bank[2] & m_mmc_prg_bank[3] remain always the same in most MMC3 variants
 	m_mmc_prg_bank[1] = m_mmc_prg_bank[3] = 0xfff; // but some pirate clone mappers change them after writing certain registers
-	std::fill(std::begin(m_mmc_vrom_bank), std::end(m_mmc_vrom_bank), 0x00);
+	m_mmc_vrom_bank[0] = 0;
+	m_mmc_vrom_bank[1] = 2;
+	m_mmc_vrom_bank[2] = 4;
+	m_mmc_vrom_bank[3] = 5;
+	m_mmc_vrom_bank[4] = 6;
+	m_mmc_vrom_bank[5] = 7;
+	m_mmc_vrom_bank[6] = 0;
+	m_mmc_vrom_bank[7] = 0;
```

Comment:
Do all MMC3 chips start with this magic initial banking arrangement, or is it just some specific pirate carts?

Reply:
Probably not. The problem is that this class, nes_txrom_device, is both supposed to represent the family of related TxROM Nintendo boards with their real MMC3 chips but is also used as the abstract MMC3 emulation to now 60 some other board types (no, most are not mine 😃 ). There's a lot of generalization and generic support for those other clone chips as a result. For instance, even right here regs 6 and 7 represent registers that don't even exist on an official MMC3 chip and are only used in subclasses (I have no idea which and haven't studied the code which is why I've erred at keeping them 0 here).

Furthermore, I couldn't tell you with confidence what, if any, these values should be even for a better-known chip like a real MMC3. For PRG, I *think* what would be most accurate would be to leave everything except the reset vector (last 8k) bank undefined and untouched at power-on and reset. For VROM, maybe it's similarly undefined? Dunno.

Anyway, without more info, to me the above arrangement seems no more magical than zeroing everything. At least SOME board boots this way. In the end though, either way is obviously too tidy to be the real behavior of each and every MMC3 variant.